x-pack/plugins/enterprise_search/public/applications/app_search/components/analytics/analytics_logic.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+import queryString from 'query-string';
+
+import { KibanaLogic } from '../../../shared/kibana';
+import { HttpLogic } from '../../../shared/http';
+import { flashAPIErrors } from '../../../shared/flash_messages';
+import { EngineLogic } from '../engine';
+
+import { AnalyticsData, QueryDetails } from './types';
+
+export interface AnalyticsValues extends AnalyticsData, QueryDetails {
+  dataLoading: boolean;
+}
+
+export interface AnalyticsActions {
+  onAnalyticsUnavailable(): void;
+  onAnalyticsDataLoad(data: AnalyticsData): AnalyticsData;
+  onQueryDataLoad(data: QueryDetails): QueryDetails;
+  loadAnalyticsData(): void;
+  loadQueryData(query: string): string;
+}
+
+export const AnalyticsLogic = kea<MakeLogicType<AnalyticsValues, AnalyticsActions>>({
+  path: ['enterprise_search', 'app_search', 'analytics_logic'],
+  actions: () => ({
+    onAnalyticsUnavailable: true,
+    onAnalyticsDataLoad: (data) => data,
+    onQueryDataLoad: (data) => data,
+    loadAnalyticsData: true,
+    loadQueryData: (query) => query,
+  }),
+  reducers: () => ({
+    dataLoading: [
+      true,
+      {
+        loadAnalyticsData: () => true,
+        loadQueryData: () => true,
+        onAnalyticsUnavailable: () => false,
+        onAnalyticsDataLoad: () => false,
+        onQueryDataLoad: () => false,
+      },
+    ],
+    analyticsUnavailable: [
+      false,
+      {
+        onAnalyticsUnavailable: () => true,
+        onAnalyticsDataLoad: (_, { analyticsUnavailable }) => analyticsUnavailable,
+        onQueryDataLoad: (_, { analyticsUnavailable }) => analyticsUnavailable,
+      },
+    ],
+  }),
+  listeners: ({ actions }) => ({
+    loadAnalyticsData: async () => {
+      const { history } = KibanaLogic.values;
+      const { http } = HttpLogic.values;
+      const { engineName } = EngineLogic.values;
+
+      try {
+        const { start, end, tag } = queryString.parse(history.location.search);
+        const query = { start, end, tag, size: 20 };
+        const url = `/api/app_search/engines/${engineName}/analytics/queries`;
+
+        const response = await http.get(url, { query });
+
+        if (response.analyticsUnavailable) {
+          actions.onAnalyticsUnavailable();
+        } else {
+          actions.onAnalyticsDataLoad(response);
+        }
+      } catch (e) {
+        flashAPIErrors(e);
+        actions.onAnalyticsUnavailable();
+      }
+    },
+    loadQueryData: async (query) => {
```

Comment:
loadAnalyticsData and loadQueryData are fairly similar in structure - I debated trying to DRY them out (a shared fn that gets passed an API URL/query/callback), but in the end decided it wasn't worth it. We likely aren't ever going to have more than just these 2 API calls so I prioritized straightforward code over more abstract

Reply:
+1 I think that was a good choice